### DISCUSSION TITLE
What is the best practice for specifying the MSVC target platform

### DISCUSSION THREAD

**Discussion by Demonese** [mesonbuild/meson#12069] [Q&A] [answered]
To build x86 executable on x86_64 machine, CMake provide a easy way: `-G "Visual Studio 17 2022" -A Win32`.

But meson doesn't seem to provide similar way, instead cross_file.

I write cross_file.txt:
```
[host_machine]
system = 'windows'
cpu_family = 'x86'
cpu = 'i686'
endian = 'little'
```

and run meson:
```
meson setup --backend=vs2022 --cross-file cross_file.txt
```

It tells me no C compiler specified. I have specified the backend as vs, why? 
```
The Meson build system
Version: 1.1.1
Source dir: ...
Build dir: ...
Build type: cross build
Project name: ...
Project version: ...
Activating VS 17.6.5

...\meson.build:1:0: ERROR: 'c' compiler binary not defined in cross or native file
```

So what's the best practice?

**Comment by xclaesse** [comment, marked as answer]
I don't think MSVC can cross compile, you need to activate the x86 env, Simply run Meson inside the `x86 Native Tools Command Prompt for VS 2022` terminal instead of `x64 Native Tools Command Prompt for VS 2022`.

Meson has the [`--vsenv`](https://mesonbuild.com/Builtin-options.html#details-for-vsenv) argument to enable the MSVC env for you (that's the default when using `vs` backend) but that can only activate the x64 env currently. I would like to add `--vsenv x86` syntax at some point.
